stdlib/Distributed/src/managers.jl

```diff
@@ -413,13 +413,14 @@ end
 """
     addprocs(; kwargs...) -> List of process identifiers
 
-Equivalent to `addprocs(Sys.CPU_THREADS; kwargs...)`
+Equivalent to `addprocs(Sys.CPU_THREADS; kwargs...)` if processes have not been launched,
+otherwise it will add only the remaining processes to reach Sys.CPU_THREADS.
 
 Note that workers do not run a `.julia/config/startup.jl` startup script, nor do they synchronize
 their global state (such as global variables, new method definitions, and loaded modules) with any
 of the other running processes.
 """
-addprocs(; kwargs...) = addprocs(Sys.CPU_THREADS; kwargs...)
+addprocs(; kwargs...) = nprocs() < Sys.CPU_THREADS && addprocs(Sys.CPU_THREADS-nprocs()+1; kwargs...)
```

Comment:
Maybe it should return `Int[]` instead of `false` if no processors are added?

Reply:
Yes definitely. Should the check also be `<= Sys.CPU_THREADS`, since the goal is to end up with CPU_THREADS+1 processes?